org.lflang/src/org/lflang/generator/c/CCompiler.java

```diff
@@ -0,0 +1,189 @@
+/*************
+ * Copyright (c) 2019-2021, The University of California at Berkeley.
+
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND 
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES 
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON 
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***************/
+
+package org.lflang.generator.c;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+import org.lflang.ErrorReporter;
+import org.lflang.FileConfig;
+import org.lflang.Mode;
+import org.lflang.TargetConfig;
+import org.lflang.generator.GeneratorCommandFactory;
+import org.lflang.util.LFCommand;
+
+/**
+ * Responsible for creating and executing the necessary native command to compile code that is generated
+ * by the CGenerator. This class invokes a compiler directly.
+ * 
+ * @author Soroush Bateni <soroush@utdallas.edu>
```

Comment:
I'm glad that we're factoring things out of `GeneratorBase`, but we should be mindful of the authorship of code that gets factor out; if this code gets put in a different class, then the original authors should also be listed as a coauthor of the new class. Granted, it's sometimes a bit difficult to determine who originally authored code, but we should try to be mindful of this. This is a general comment that doesn't only apply to this class, of course.

Reply:
This `@author` thing is something that Eclipse adds for me automatically whenever I create a Java class. Perhaps authors of the original file (`GeneratorBase`) in this case should automatically also be included here in the same order?

Reply:
If in doubt, `git blame` can also be a useful tool.

Reply:
Okay, thanks for the pointer!